pallets/parachain-staking/src/lib.rs

```diff
@@ -1502,6 +1476,55 @@ pub mod pallet {
 			<SelectedCandidates<T>>::get().binary_search(acc).is_ok()
 		}
 
+		pub fn join_candidates_inner(
+			acc: T::AccountId,
+			bond: BalanceOf<T>,
+			candidate_count: u32,
+			skip_bond_check: bool,
+		) -> DispatchResultWithPostInfo {
+			ensure!(!Self::is_candidate(&acc), Error::<T>::CandidateExists);
+			ensure!(!Self::is_delegator(&acc), Error::<T>::DelegatorExists);
+			ensure!(
+				skip_bond_check || bond >= T::MinCandidateStk::get(),
```

Comment:
It should be better to move this check inside that `join_candidates` call to simplify the check (avoid the boolean)

Reply:
good catch! :thumbsup: